contracts/currency/currency.abi

```diff
@@ -1,6 +1,6 @@
 {
   "types": [{
-      "new_type_name": "account_name",
+      "type_name": "account_name",
```

Comment:
`"new_type_name"` is correct. See `libraries/types/types.eos` where `abi` is defined as:
```
struct abi
   types   type_def[]
   structs struct_t[]
   actions action[]
   tables  table[]
```
And `type_def` is defined as:
```
struct type_def
   new_type_name type_name
   type        type_name
```
Which generates (libraries/types/include/eos/types/generated.hpp):
```
    struct abi { 
        abi() = default;
        abi(const vector<type_def>& types, const vector<struct_t>& structs, const vector<action>& actions, const vector<table>& tables)
           : types(types), structs(structs), actions(actions), tables(tables) {}

        vector<type_def>                 types;
        vector<struct_t>                 structs;
        vector<action>                   actions;
        vector<table>                    tables;
    };
```
```
    struct type_def { 
        type_def() = default;
        type_def(const type_name& new_type_name, const type_name& type)
           : new_type_name(new_type_name), type(type) {}

        type_name                        new_type_name;
        type_name                        type;
    };
```